services/app-web/src/pages/StateSubmission/RateDetails/V2/RateDetailsV2.tsx

```diff
@@ -0,0 +1,375 @@
+import React, { useEffect } from 'react'
+import { Form as UswdsForm } from '@trussworks/react-uswds'
+import { Formik, FormikErrors } from 'formik'
+import { useNavigate, useParams } from 'react-router-dom'
+import { v4 as uuidv4 } from 'uuid'
+
+import styles from '../../StateSubmissionForm.module.scss'
+
+import { ErrorSummary } from '../../../../components'
+import { RateDetailsFormSchema } from '../RateDetailsSchema'
+import { PageActions } from '../../PageActions'
+
+import {
+    formatActuaryContactsForForm,
+    formatDocumentsForForm,
+    formatDocumentsForGQL,
+    formatForForm,
+    formatFormDateForGQL,
+} from '../../../../formHelpers/formatters'
+import { useS3 } from '../../../../contexts/S3Context'
+import { S3ClientT } from '../../../../s3'
+import {
+    FileItemT,
+    isLoadingOrHasFileErrors,
+} from '../../../../components/FileUpload'
+import { RouteT, RoutesRecord } from '../../../../constants'
+import {
+    Rate,
+    RateFormDataInput,
+    RateRevision,
+} from '../../../../gen/gqlClient'
+import { SingleRateCertV2 } from './SingleRateCertV2'
+import type { SubmitOrUpdateRate } from '../../../RateEdit/RateEdit'
+
+export type RateDetailFormValues = {
+    id: Rate['id']
+    rateType: RateRevision['formData']['rateType']
+    rateCapitationType: RateRevision['formData']['rateCapitationType']
+    rateDateStart: RateRevision['formData']['rateDateStart']
+    rateDateEnd: RateRevision['formData']['rateDateEnd']
+    rateDateCertified: RateRevision['formData']['rateDateCertified']
+    effectiveDateStart: RateRevision['formData']['amendmentEffectiveDateStart']
+    effectiveDateEnd: RateRevision['formData']['amendmentEffectiveDateEnd']
+    rateProgramIDs: RateRevision['formData']['rateProgramIDs']
+    rateDocuments: FileItemT[]
+    supportingDocuments: FileItemT[]
+    actuaryContacts: RateRevision['formData']['certifyingActuaryContacts']
+    addtlActuaryContacts: RateRevision['formData']['addtlActuaryContacts']
+    actuaryCommunicationPreference: RateRevision['formData']['actuaryCommunicationPreference']
+    packagesWithSharedRateCerts: RateRevision['formData']['packagesWithSharedRateCerts']
+}
+
+// We have a list of rates to enable multi-rate behavior
+export type RateDetailFormConfig = {
+    rates: RateDetailFormValues[]
+}
+
+const generateFormValues = (
+    getKey: S3ClientT['getKey'],
+    rateRev?: RateRevision,
+    rateID?: string
+): RateDetailFormValues => {
+    const rateInfo = rateRev?.formData
+    const newRateID = uuidv4()
+
+    return {
+        id: rateID ?? newRateID,
```

Comment:
I think we don't need to generate IDs here anymore, the API can generate an id on creation now that rates aren't stuck inside contracts

Reply:
@macrael Great, I would love to do that now. Next steps I see  - I adjust createRate api to also generate ids when they are missing and remove id generation on frontend when the flag is on. I'm assuming we still want to generate ids on frontend when flag is off since that's using protos. Let me know if I missed anything about what this change entails.

Reply:
Yes I think that's right. If the feature flag is off, rates should still be being created in the call to updateHPFormData so they would need the IDs embedded, but that's not using rateDetailsV2 right?

Reply:
yup yup you right. I made it too complicated. No createRate yet and I made an entirely new component so no feature flag. Oops had some Friday brain going on!